engine/glfw/lib/x11/x11_joystick.c

```diff
@@ -296,6 +300,15 @@ static void pollJoystickEvents( void )
             // Read all queued events (non-blocking)
             while( read(_glfwJoy[i].fd, &e, sizeof(struct js_event)) > 0 )
             {
+                if (_glfwJoyConnected[ i ] == 0)
+                {
+                    _glfwJoyConnected[ i ] = 1;
+                    if (_glfwWin.gamepadCallback)
+                    {
+                        _glfwWin.gamepadCallback(_glfwJoy[ i ].InputId, 1);
+                    }
+                }
```

Comment:
In the update loop for the newer api (`/dev/input/js`), we never checked for a connection and thus never called the callback.